src/include/duckdb.h

```diff
@@ -308,7 +308,8 @@ typedef enum { DuckDBSuccess = 0, DuckDBError = 1 } duckdb_state;
 typedef enum {
 	DUCKDB_PENDING_RESULT_READY = 0,
 	DUCKDB_PENDING_RESULT_NOT_READY = 1,
-	DUCKDB_PENDING_ERROR = 2
+	DUCKDB_PENDING_ERROR = 2,
+	DUCKDB_PENDING_ALL_TASKS_BLOCKED = 3
```

Comment:
Nit: `ALL_TASKS_BLOCKED` could be a confusing name, as what actually happens is that there is just no tasks for the current executor to be picked up. There can still be background workers executing in-progress tasks (and even possibly enqueue more tasks)